python/lsst/ci/build.py

```diff
@@ -328,15 +333,60 @@ def write_status(self):
             yaml.dump(status, sf, encoding="utf-8", default_flow_style=False)
 
     @staticmethod
-    def run(args):
+    def get_agent() -> str:
+        """Determine the system agent based on OS type and architecture.
+
+        Returns
+        -------
+        `str`
+            The formatted agent string, such as `"macos_arm64"` or
+            `"linux_x86_64"`. If the system platform or architecture.
+
+        Raises
+        ------
+        `UnknownAgentError`
+            If either the operating-system name or the CPU architecture
+            cannot be obtained.
+        """
+        os_type = sys.platform
+        arch = platform.machine()
+
+        if not os_type or not arch:
+            raise UnknownAgentError("unknown_agent")
+
+        match os_type:
+            case "darwin":
+                agent = f"macos_{arch}"
+            case _:
+                agent = f"{os_type}_{arch}"
+
+        # Print verification message
+        print(f"Agent verified: {agent}")
+
+        return agent
+
+    @staticmethod
+    def run(args) -> None:
+        """Run the build process, including GitHub status updates."""
+        # agent = Builder.get_agent()
+
         # Ensure build directory exists and is writable
         build_dir = args.build_dir
         if not os.access(build_dir, os.W_OK):
             raise Exception(f"Directory {build_dir!r} does not exist or isn't writable.")
 
+        # Load PR pydantic model created in prepare.py
+        manager = PRInfoManager(build_dir)
+        pr_info = manager.load_pr_info()
+        # pr_info = Builder.load_pr_info(build_dir)
+
+        if not pr_info:
+            print("WARNING: PR information could not be loaded. Skipping posting Jenkins statuses on GitHub.")
+
+        post_statuses = PostStatuses()
```

Comment:
Suggest this class disappears. If it stays consider making the two methods class methods since they have no state to store.